scripts/sanitycheck.fsx

```diff
@@ -0,0 +1,592 @@
+#!/usr/bin/env -S dotnet fsi
+
+open System
+open System.IO
+open System.Linq
+open System.Diagnostics
+
+open System.Text
+open System.Text.RegularExpressions
+
+#r "System.Core.dll"
+open System.Xml
+#r "System.Xml.Linq.dll"
+
+open System.Xml.Linq
+open System.Xml.XPath
+
+#r "nuget: Fsdk, Version=0.6.0--date20230214-0422.git-1ea6f62"
+
+open Fsdk
+open Fsdk.Process
+
+
+#r "nuget: Microsoft.Build, Version=16.11.0"
+open Microsoft.Build.Construction
+let ScriptsDir = __SOURCE_DIRECTORY__ |> DirectoryInfo
+let RootDir = Path.Combine(ScriptsDir.FullName, "..") |> DirectoryInfo
+
+let NugetSolutionPackagesDir =
+    Path.Combine(RootDir.FullName, "packages") |> DirectoryInfo
+
+
+module MapHelper =
+    let GetKeysOfMap(map: Map<'K, 'V>) : seq<'K> =
+        map |> Map.toSeq |> Seq.map fst
+
+    let MergeIntoMap<'K, 'V when 'K: comparison>
+        (from: seq<'K * 'V>)
+        : Map<'K, seq<'V>> =
+        let keys = from.Select(fun (k, v) -> k)
+
+        let keyValuePairs =
+            seq {
+                for key in keys do
+                    let valsForKey =
+                        (from.Where(fun (k, v) -> key = k))
+                            .Select(fun (k, v) -> v)
+                        |> seq
+
+                    yield key, valsForKey
+            }
+
+        keyValuePairs |> Map.ofSeq
+
+[<StructuralEquality; StructuralComparison>]
+type private PackageInfo =
+    {
+        PackageId: string
+        PackageVersion: string
+        ReqReinstall: Option<bool>
+    }
+
+type private DependencyHolder =
+    {
+        Name: string
+    }
+
+[<CustomComparison; CustomEquality>]
+type private ComparableFileInfo =
+    {
+        File: FileInfo
+    }
+
+    member self.DependencyHolderName: DependencyHolder =
+        if self.File.FullName.ToLower().EndsWith ".nuspec" then
+            {
+                Name = self.File.Name
+            }
+        else
+            {
+                Name = self.File.Directory.Name + "/"
+            }
+
+    interface IComparable with
+        member this.CompareTo obj =
+            match obj with
+            | null -> this.File.FullName.CompareTo null
+            | :? ComparableFileInfo as other ->
+                this.File.FullName.CompareTo other.File.FullName
+            | _ -> invalidArg "obj" "not a ComparableFileInfo"
+
+    override this.Equals obj =
+        match obj with
+        | :? ComparableFileInfo as other ->
+            this.File.FullName.Equals other.File.FullName
+        | _ -> false
+
+    override this.GetHashCode() =
```

Comment:
@Mersho please rename all occurrences of `this` to `self` to comply with our F# coding guidelines